### PR TITLE
Localized Message Source guide

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Comment by sdelamo** [comment]
@kroegerk can you proofread this?

File: guides/localized-message-source/java/src/main/java/example/micronaut/MessageSourceFactory.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import io.micronaut.context.MessageSource;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.i18n.ResourceBundleMessageSource;
+import jakarta.inject.Singleton;
+
+@Factory // <1>
+class MessageSourceFactory {
+    @Singleton // <2>
+    MessageSource createMessageSource() {
+        return new ResourceBundleMessageSource("i18n.messages");
```
Comment (burtbeckwith) [inline guides/localized-message-source/java/src/main/java/example/micronaut/MessageSourceFactory.java:12]: should have a callout here explaining "i18n.messages"

**Review by burtbeckwith — APPROVED** [review]
looks good, just needs an intro with an overview of i18n and some info about support in Micronaut

File: guides/localized-message-source/localized-message-source.adoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+callout:micronaut-test[1]
+callout:http-client[2]
+callout:http-request[3]
+
+common:testApp.adoc[]
+
+== Learn More
+
+The application changes the locale with `Accept-Language` HTTP Header. Learn more about https://docs.micronaut.io/latest/guide/#localeResolution[Locale Resolution].
```
Comment (kroegerk) [inline guides/localized-message-source/localized-message-source.adoc]: I think this is missing an article, but I'm not sure which is more appropriate.

```suggestion
The application changes the locale with an `Accept-Language` HTTP Header. Learn more about https://docs.micronaut.io/latest/guide/#localeResolution[Locale Resolution].
```

OR

```suggestion
The application changes the locale with the `Accept-Language` HTTP Header. Learn more about https://docs.micronaut.io/latest/guide/#localeResolution[Locale Resolution].
```

**Pull request merged**